lib/recode_url.rb

```diff
@@ -0,0 +1,10 @@
+class RecodeUrl
+  def initialize(app, options = {})
+    @app = app
+  end
+  
+  def call(env)
+    env['PATH_INFO'] = env['PATH_INFO'].gsub('/accounts','') if env['REQUEST_METHOD'] == 'GET'
+    @app.call(env)
+  end
+end
```

Comment:
you could use the bang version of gsub that modifies in place, but probably should also check that PATH_INFO exists:
```ruby
env['PATH_INFO'].gsub!('/accounts','') if env['REQUEST_METHOD'] == 'GET' && env['PATH_INFO'].present?
```

Also this is completely subjective but I prefer putting the constant on the left so that it'll break if you use a single equal sign instead of accidentally assigning.
```ruby
if 'GET' == env['REQUEST_METHOD']
```

Reply:
+1 on `if constant == variable` for safety

Reply:
I think the gsub line will incorrectly change a path of `/something/accounts/blah` to `/something/blah`.  Can you add a spec that shows this happens?  Then change the gsub command to use a regular expression in the first argument that requires the `/accounts` to be at the start of the path.  rubular.com is a great tool to play with for this btw.